src/Common/Key_Value_Cache/Key_Value_Cache_Table.php

```
@@ -0,0 +1,215 @@
+					Schema::table_name(),
+					$key,
+					$current_time
+				)
+			);
+		} catch ( Exception $e ) {
+			return false;
+		}
+
+		return $exists > 0;
+	}
+
+	/**
+	 * Returns a cached value if it exists.
+	 *
+	 * Note the method does not inquire into the nature of the stored value, it's returned as it is and no guarantee
+	 * is made on the integrity or correctness of the value.
+	 *
+	 * @since TBD
+	 *
+	 * @param string      $key      The key to return the value for.
+	 * @param string|null $fallback The fallback value to return if the key does not exist.
+	 *
+	 * @return string The cached value or the default value if not set.
+	 */
+	public function get( string $key, string $fallback = '' ): string {
+		$current_time = time();
+
+		try {
+			$value = DB::get_var(
+				DB::prepare(
+					'SELECT `value` FROM %i WHERE `cache_key` = %s AND (`expiration` = 0 OR `expiration` > %d)',
+					Schema::table_name(),
+					$key,
+					$current_time
+				)
+			);
+		} catch ( Exception $e ) {
+			return $fallback;
+		}
+
+		if ( null === $value ) {
+			return $fallback;
+		}
+
+		return (string) $value;
+	}
+
+	/**
+	 * Stores a value for a key.
+	 *
+	 * The maximum length of 191 chars aligns with the table definition and is the same length
+	 * of the index used in the `wp_postmeta` table for meta_keys.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $key   The key to store the value for.
+	 * @param string $value The value to store for the key.
+	 * @param int    $expiration The cache expiration, it cannot be below 300 seconds.
+	 *
+	 * @return bool Whether the value was correctly stored or not.
+	 */
+	public function set( string $key, string $value, int $expiration = 300 ): bool {
+		// Bail if the key is longer than 191 characters.
+		if ( strlen( $key ) > 191 ) {
+			return false;
+		}
+
+		if ( $expiration < 300 ) {
+			// To stick with WP VIP min requirements, we're not going to store values with an expiration under 5 minutes.
+			return false;
+		}
```

The silent failure is the desired approach ?